```
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Core.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_Core-7ddc39.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Core.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Core-7ddc39.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Core.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Core-7ddc39.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Serial.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_Serial-1710de.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Serial.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Serial-1710de.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Serial.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Serial-1710de.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_SYCL-c6298b.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SYCL-c6298b.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SYCL-c6298b.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL_Instance.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_SYCL_Instance-958be0.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL_Instance.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SYCL_Instance-958be0.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SYCL_Instance.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SYCL_Instance-958be0.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Error.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_Error-6b6ab3.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Error.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Error-6b6ab3.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_Error.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_Error-6b6ab3.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SharedAlloc.cpp.o): in function `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)':
Kokkos_SharedAlloc-b56895.cpp:(.text+0x0): multiple definition of `Kokkos::Impl::impl_test_fallback_bhalf(Kokkos::Experimental::Impl::floating_point_wrapper<sycl::_V1::ext::oneapi::bfloat16>)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x0): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SharedAlloc.cpp.o): in function `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SharedAlloc-b56895.cpp:(.text+0x10): multiple definition of `Kokkos::PerTeam(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x10): first defined here
/usr/bin/ld: ../src/libkokkoscore.a(Kokkos_SharedAlloc.cpp.o): in function `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)':
Kokkos_SharedAlloc-b56895.cpp:(.text+0x20): multiple definition of `Kokkos::PerThread(Kokkos::Impl::SYCLTeamMember const&)'; /tmp/icpx-52ec0e0767/UnitTest_DeviceAndThreads-010639.o:UnitTest_DeviceAndThreads-a97254.cpp:(.text+0x20): first defined here
Compilation from IR - skipping loading of FCL
Build succeeded.
Compilation from IR - skipping loading of FCL
icpx: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/core/src/SYCL/Kokkos_SYCL_Half_MathematicalFunctions.hpp
+++ b/core/src/SYCL/Kokkos_SYCL_Half_MathematicalFunctions.hpp
@@ -130,7 +130,7 @@ KOKKOS_SYCL_HALF_UNARY_FUNCTION(rsqrt)
         Experimental::bhalf_t::impl_type(x));      \
   }
 
-Kokkos::Experimental::bhalf_t impl_test_fallback_bhalf(
+inline Kokkos::Experimental::bhalf_t impl_test_fallback_bhalf(
     Kokkos::Experimental::bhalf_t) {
   return Kokkos::Experimental::bhalf_t(0.f);
 }

--- a/core/src/SYCL/Kokkos_SYCL_Team.hpp
+++ b/core/src/SYCL/Kokkos_SYCL_Team.hpp
@@ -473,12 +473,12 @@ ThreadVectorRange(const Impl::SYCLTeamMember& thread, iType1 arg_begin,
       thread, iType(arg_begin), iType(arg_end));
 }
 
-Impl::ThreadSingleStruct<Impl::SYCLTeamMember> PerTeam(
+inline Impl::ThreadSingleStruct<Impl::SYCLTeamMember> PerTeam(
     const Impl::SYCLTeamMember& thread) {
   return Impl::ThreadSingleStruct<Impl::SYCLTeamMember>(thread);
 }
 
-Impl::VectorSingleStruct<Impl::SYCLTeamMember> PerThread(
+inline Impl::VectorSingleStruct<Impl::SYCLTeamMember> PerThread(
     const Impl::SYCLTeamMember& thread) {
   return Impl::VectorSingleStruct<Impl::SYCLTeamMember>(thread);
 }

